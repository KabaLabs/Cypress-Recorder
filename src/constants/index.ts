export enum EventType {
  CLICK = 'click',
  CHANGE = 'change',
  DBCLICK = 'dbclick',
  KEYDOWN = 'keydown',
  SUBMIT = 'submit',
}

export enum ControlAction {
  START = 'startRec',
  STOP = 'stopRec',
  RESET = 'resetRec',
  DELETE = 'deleteBlock',
}
