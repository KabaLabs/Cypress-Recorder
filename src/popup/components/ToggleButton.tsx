import * as React from 'react';
import { ControlAction } from '../../constants';

export interface ToggleButtonProps {
  isValidTab: boolean,
  recStatus: string,
  handleToggle: (action: ControlAction) => void,
}

export default ({ recStatus, handleToggle, isValidTab }: ToggleButtonProps) => {
  const handleClick = (): void => {
    let action: ControlAction;
    if (recStatus === 'off' || recStatus === 'paused') action = ControlAction.START;
    else if (recStatus === 'on') action = ControlAction.STOP;
    handleToggle(action);
  };

  const buttonClass: string = (!isValidTab && (recStatus === 'off' || recStatus === 'paused')) ? 'disabled-button' : 'button';

  return (
    <>
      <button type="button" className={buttonClass} onClick={handleClick} disabled={!isValidTab && (recStatus === 'off' || recStatus === 'paused')}>
        {(recStatus === 'off' || recStatus === 'paused') && !isValidTab && 'Invalid Tab'}
        {recStatus === 'off' && isValidTab && 'Start Recording'}
        {recStatus === 'paused' && isValidTab && 'Resume'}
        {recStatus === 'on' && 'Stop Recording'}
      </button>
    </>
  );
};
