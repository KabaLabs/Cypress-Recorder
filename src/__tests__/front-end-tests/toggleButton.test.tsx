import * as React from 'react';
import { shallow } from 'enzyme';
import ToggleButton from '../../popup/components/ToggleButton';
import '../../setupTests';
import { ControlAction } from '../../constants';

describe('toggle button', () => {
  let wrapper;
  let handleToggleMock;
  beforeEach(() => {
    handleToggleMock = jest.fn();
  });
  it('should not call handletoggle when clicked and recStatus is "off" and tab is invalid', () => {
    wrapper = shallow(<ToggleButton recStatus='off' isValidTab={false} handleToggle={handleToggleMock}/>)
    expect(wrapper).toMatchSnapshot();
    const button = wrapper.find('button');
    expect(button.text()).toBe('Invalid Tab');
    expect(button.prop('disabled')).toBe(true);
  });
  it('should call handletoggle with argument "startRec" when clicked and recStatus is "off" and tab is valid', () => {
    wrapper = shallow(<ToggleButton recStatus='off' isValidTab={true} handleToggle={handleToggleMock}/>)
    expect(wrapper).toMatchSnapshot();
    const button = wrapper.find('button');
    button.simulate('click');
    expect(handleToggleMock).toHaveBeenCalledWith(ControlAction.START);
  });
  it('should call handletoggle with argument "stopRec" when clicked and recStatus is "on"', () => {
    wrapper = shallow(<ToggleButton recStatus='on' isValidTab={true} handleToggle={handleToggleMock}/>)
    expect(wrapper).toMatchSnapshot();
    const button = wrapper.find('button');
    button.simulate('click');
    expect(handleToggleMock).toHaveBeenCalledWith(ControlAction.STOP);
  });
  it('should call handletoggle with argument "startRec" when clicked and recStatus is "paused"', () => {
    wrapper = shallow(<ToggleButton recStatus='paused' isValidTab={true} handleToggle={handleToggleMock}/>)
    expect(wrapper).toMatchSnapshot();
    const button = wrapper.find('button');
    button.simulate('click');
    expect(handleToggleMock).toHaveBeenCalledWith(ControlAction.START);
  });
})
