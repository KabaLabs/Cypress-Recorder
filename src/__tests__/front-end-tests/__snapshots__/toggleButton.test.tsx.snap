// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toggle button should call handletoggle with argument "startRec" when clicked and recStatus is "off" and tab is valid 1`] = `
<Fragment>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Start Recording
  </button>
</Fragment>
`;

exports[`toggle button should call handletoggle with argument "startRec" when clicked and recStatus is "paused" 1`] = `
<Fragment>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Resume
  </button>
</Fragment>
`;

exports[`toggle button should call handletoggle with argument "stopRec" when clicked and recStatus is "on" 1`] = `
<Fragment>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Stop Recording
  </button>
</Fragment>
`;

exports[`toggle button should not call handletoggle when clicked and recStatus is "off" and tab is invalid 1`] = `
<Fragment>
  <button
    className="disabled-button"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    Invalid Tab
  </button>
</Fragment>
`;
